src/groups/mqb/mqbauthn/mqbauthn_basicauthenticator.cpp

```diff
@@ -0,0 +1,218 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mqbauthn_basicauthenticator.h>
+
+// MQB
+#include <mqbcfg_messages.h>
+#include <mqbplug_authenticator.h>
+
+// BDE
+#include <bdlt_timeunitratio.h>
+#include <bsl_iostream.h>
+#include <bsl_memory.h>
+#include <bsl_optional.h>
+#include <bsl_string.h>
+#include <bsl_string_view.h>
+#include <bslma_allocator.h>
+#include <bslma_managedptr.h>
+#include <bsls_assert.h>
+
+namespace BloombergLP {
+namespace mqbauthn {
+
+namespace {
+
+const int k_AUTHN_DURATION_SECONDS = 600;
+
+}  // namespace
+
+// -------------------------------
+// class BasicAuthenticationResult
+// -------------------------------
+
+BasicAuthenticationResult::BasicAuthenticationResult(
+    bsl::string_view   principal,
+    bsls::Types::Int64 lifetimeMs,
+    bslma::Allocator*  allocator)
+: d_principal(principal, allocator)
+, d_lifetimeMs(lifetimeMs)
+{
+}
+
+BasicAuthenticationResult::~BasicAuthenticationResult()
+{
+}
+
+bsl::string_view BasicAuthenticationResult::principal() const
+{
+    return d_principal;
+}
+
+const bsl::optional<bsls::Types::Int64>&
+BasicAuthenticationResult::lifetimeMs() const
+{
+    return d_lifetimeMs;
+}
+
+// ------------------------
+// class BasicAuthenticator
+// ------------------------
+
+BasicAuthenticator::BasicAuthenticator(
+    const mqbcfg::AuthenticatorPluginConfig* config,
+    bslma::Allocator*                        allocator)
+: d_allocator_p(allocator)
+, d_credentials(allocator)
+, d_isStarted(false)
+{
+    if (!config) {
+        BALL_LOG_WARN << "No config provided for BasicAuthenticator '"
+                      << name() << "' with mechanism '" << mechanism() << "'.";
+        return;
+    }
+
+    // Load the configured key-values as username and password
+    bool isValueFound = false;
+    bsl::vector<mqbcfg::PluginSettingKeyValue>::const_iterator it =
+        config->settings().cbegin();
+    for (; it != config->settings().cend(); ++it) {
+        if (!it->value().isStringValValue()) {
+            BALL_LOG_WARN << "Expected string for 'key-value' setting...";
+            continue;
+        }
+        d_credentials[it->key()] = it->value().stringVal();
+        isValueFound             = true;
+    }
+    if (isValueFound) {
+        BALL_LOG_INFO << "... setting found.  Loaded " << d_credentials.size()
+                      << " credential(s).";
+    }
+    else {
+        BALL_LOG_INFO << "... setting not found.  Make sure this is expected.";
```

Comment:
```suggestion
        BALL_LOG_WARN << "No credentials found in configuration. Please double check if this is intended."
```
Can log a similar message to https://github.com/bloomberg/blazingmq/blob/157313364b9eb131df3022f1462834c0f609fcf3/src/groups/mqb/mqbplug/mqbplug_pluginmanager.cpp#L320